### PR TITLE
Add --all to remove containers

### PR DESCRIPTION
**Pull request by TomSweeneyRedHat**
Signed-off-by: TomSweeneyRedHat <tsweeney@redhat.com>

Add the --all flag to buildah rm.  Will follow up for images in a separate patch/pr.  Addresses at least part of #374

### CODE REVIEW

File: cmd/buildah/rm.go [hunk 1]
```diff
@@ -31,27 +39,45 @@ func rmCmd(c *cli.Context) error {
...
+			id := builder.ContainerID
+			err = builder.Delete()
+			if e == nil {
+				e = err
+			}
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "error removing container %q: %v\n", builder.Container, err)
+				continue
+			}
+			fmt.Printf("%s\n", id)
```
Comment (TomSweeneyRedHat) [inline cmd/buildah/rm.go]: diff wasn't very kind here.  The code in the else was all the pre-existing code and didn't change except for an extra leading tab.

**Comment by nalind** [comment]
LGTM

File: cmd/buildah/rm.go [hunk 2]
```diff
@@ -31,27 +38,45 @@ func rmCmd(c *cli.Context) error {
 	}
 
 	var e error
-	for _, name := range args {
-		builder, err := openBuilder(store, name)
-		if e == nil {
-			e = err
-		}
+	if c.Bool("all") {
+		builders, err := openBuilders(store)
```
Comment (nalind) [inline cmd/buildah/rm.go]: This returns the set of containers which buildah is using, but would not include those which it is not (for example, any created by cri-o).  Is this the intention?

Comment (TomSweeneyRedHat) [reply cmd/buildah/rm.go]: I thought it was just to do Buildah containers.  @rhatdan?

Comment (TomSweeneyRedHat) [reply cmd/buildah/rm.go]: That is I meant it to remove the containers shown with "buildah containers", holler if this doesn't hit that mark, seems to in my testing, but I didn't try adding a container via Podman in my tests.

Comment (rhatdan) [reply cmd/buildah/rm.go]: Yes we need to be careful with this going forward,  We have to figure out when we merge cri-o/podman/buildah together, what is the expected output of this.

If buildah images

List all images, which I think it should eventually.  I would want buildah rmi -a to remove them all whether they are created via buildah or other tools.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

File: cmd/buildah/rm.go (hunk 2 above)
Comment (nalind) [reply cmd/buildah/rm.go]: We set a `*` in the `BUILDER` column in the `buildah containers` output if the container has some buildah-specific data associated with it, and containers which don't (i.e., created by cri-o, and not yet touched by buildah) are not listed unless `buildah containers` was passed the `-a` flag.  The two `-a` flags don't seem to work the same way.

Comment (TomSweeneyRedHat) [reply cmd/buildah/rm.go]: Hmm, sold.  I've to do rmi too, I'll test/look into the containers without the '*' tag too.

**Pull request closed without merging**